doc/developer/platform/formalism.md

```diff
@@ -747,12 +742,12 @@ The dataflow remains active until a corresponding `DROP INDEX` command is receiv
 The Adapter layer provides interactive transactions through the `BEGIN`, `COMMIT`, and `ROLLBACK` commands.
 There are various restrictions on these transactions, primarily because the Storage and Compute layers require you to durably write before you can read from them.
 This makes read-after-write transactions challenging to perform efficiently, and they are not currently supported.
-Write-after-read transactions are implemented by performing all reads at the current timestamp and then writing at a strictly later timestamp.
-Other read-only transactions may co-occur with a write-after-read transaction (logically preceding the transaction, as its writes are only visible at its write timestamp).
-Other write-only transactions may precede or follow a write-after-read transaction, but may not occur *between* the read and write times of that transaction.
-Multiple write-after-read transactions may not co-occur (their [read, write) half-open timestamp intervals must be disjoint).
+Read-then-write transactions are implemented by performing all reads at some timestamp greater than or equal to all previous timestamps and then writing at a strictly later timestamp.
+Other read-only transactions may co-occur with a read-then-write transaction (logically preceding the transaction, as its writes are only visible at its write timestamp).
+Other write-only transactions may precede or follow a read-then-write transaction, but may not occur *between* the read and write times of that transaction.
```

Comment:
Reading this and the above, co-occur wasn't clear to me (I think it has the technical meaning "can be assigned a timestamp that lies within some read-write interval), and "may precede or follow" and "occur between" was also not pinpoint clear.